### PR TITLE
Belos: Add Tpetra Orthogonalization Example

### PR DESCRIPTION
**Pull request by jennloe**
This is a simple PR to push an example requested by @mperego and the Albany team.  (See related at https://github.com/sandialabs/Albany/pull/756.) 
They needed a function (outside of any solver interface) that can orthogonalize an arbitrary `Tpetra::MultiVector` and return the orthogonalization coefficients.  This code contains such a function and an example of how to call it.     

[Note there is also a built-in set of debug code to check that the orthogonalization function is doing the right thing.  It is commented out so that the user can enable if needed.  Right now, it may crash for certain problems due to #9856.]

@trilinos/belos

### CODE REVIEW

File: packages/belos/tpetra/example/Orthog/TpetraOrthogEx.cpp [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  for(int k=1; k<numLoops; k++){
+    pastVecArrayView = arrayViewFromVector(pastVecArray);
+    vecBlock = inputVecs.subViewNonConst(Teuchos::Range1D(k*blkSize,k*blkSize + blkSize - 1));
+    coeffNorm = Teuchos::rcp( new MAT(Teuchos::View, *Coeffs, blkSize, blkSize, k*blkSize, k*blkSize)); 
+    coeffDot = Teuchos::rcp(new MAT(Teuchos::View, *Coeffs, k*blkSize, blkSize, 0, k*blkSize));
+    Array<RCP<MAT>> dotArray; //Tuechos::Array of Matrices for coeffs from dot products
+    dotArray.append(coeffDot);
+    int rank = orthoMgr->projectAndNormalize(*vecBlock, dotArray, coeffNorm, pastVecArrayView);
+    pastVecBlock = inputVecs.subViewNonConst(Teuchos::Range1D(0,k*blkSize + blkSize - 1));
+    pastVecArray.front() = pastVecBlock;
```
[the remark this answers is not in the document]
Comment (jennloe) [reply packages/belos/tpetra/example/Orthog/TpetraOrthogEx.cpp:241]: > Good question!! In the first version you saw of this code, it was working over one block at a time. But in this version, I think it is orthogonalizing against all the previous columns at once. Which do you think is better?

@hkthorn Can you verify my answer here?

[the remark this answers is not in the document]
Comment (jennloe) [reply packages/belos/tpetra/example/Orthog/TpetraOrthogEx.cpp:241]: @iyamazaki I am adding a warning that prints to the user if vectors are not full rank.

[the remark this answers is not in the document]
Comment (jennloe) [reply packages/belos/tpetra/example/Orthog/TpetraOrthogEx.cpp:241]: Wait- NVM. I'll return it to the user as suggested.

**Comment by jennloe** [comment]
Okay @iyamazaki I think I have this ready now.

**Comment by jennloe** [comment]
Looks like the only thing failing now is a MueLu test.  That shouldn't be the case; this PR doesn't touch anything in MueLu.  @trilinos/framework do you see that issue anywhere else?

**Comment by jennloe** [comment]
@iyamazaki Can you do a final review?

File: packages/belos/tpetra/example/Orthog/TpetraOrthogEx.cpp [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    std::cout << std::endl << "We have orthogonalized " << numVecs << " vectors. Numerical rank is " << rank << "." << std::endl;
+  }
+
+  //Verify Orthogonality:
+  RCP<MAT> bigDotAns  = rcp( new MAT(numVecs, numVecs));
+  MVT::MvTransMv(1.0, *X1, *X1, *bigDotAns);
+  if(proc_verbose){
+    std::cout << std::endl << "Printed dot prod matrix for verification: " << std::endl;
+    std::cout << "Should be ones on diagonal and zeros elsewhere." << std::endl;
+    bigDotAns->print(std::cout);
```
[the remark this answers is not in the document]
Comment (jennloe) [reply packages/belos/tpetra/example/Orthog/TpetraOrthogEx.cpp:152]: I can update that in a future PR.  Can you approve this one for now?

**Review by iyamazaki — APPROVED** [review]

**Pull request merged**